app/post_metadata.go

```diff
@@ -533,9 +544,14 @@ func (a *App) getLinkMetadata(requestURL string, timestamp int64, isNewPost bool
 			return nil, nil, nil, appErr
 		}
 
-		referencedTeam, appErr := a.GetTeam(referencedChannel.TeamId)
-		if appErr != nil {
-			return nil, nil, nil, appErr
+		referencedTeam := &model.Team{}
+		if referencedChannel.Type == model.ChannelTypeDirect || referencedChannel.Type == model.ChannelTypeGroup {
+			referencedTeam.Name = ""
+		} else {
+			referencedTeam, appErr = a.GetTeam(referencedChannel.TeamId)
```

Comment:
Are we doing the required authorization checks to see if the recipient is allowed to see the team?

Reply:
Yes those checks are still done as before to verify that the recipient is allowed to view the channel and team. Before the message is broadcast to each user, the metadata is removed by `sanitizePostMetadataForUserAndChannel` here: https://github.com/mattermost/mattermost-server/blob/63dfd61b96d7d719b2a97a4999813ebfaab1e8f9/app/post_metadata.go#L166

which is called from `publishWebsocketEventForPermalinkPost` here: https://github.com/mattermost/mattermost-server/blob/63dfd61b96d7d719b2a97a4999813ebfaab1e8f9/app/post.go#L752

The only modification made to this is that we now get the previewed post and channel once before sanitizing the post for each user, which should help with performance.